contractcourt/htlc_incoming_contest_resolver.go

```diff
@@ -301,13 +305,17 @@ func (h *htlcIncomingContestResolver) Resolve() (ContractResolver, error) {
 
 		resolution, err := h.Registry.NotifyExitHopHtlc(
 			h.htlc.RHash, h.htlc.Amt, h.htlcExpiry, currentHeight,
-			circuitKey, hodlChan, payload,
+			circuitKey, hodlQueue.ChanIn(), payload,
 		)
 		if err != nil {
 			return nil, err
 		}
 
-		defer h.Registry.HodlUnsubscribeAll(hodlChan)
+		defer func() {
+			hodlQueue.Stop()
```

Comment:
Wouldn't we need to stop the queue _before_ unsubscribing? Because it looks like `HodlUnsubscribeAll` would write to the queue but then the events would never be processed?

Reply:
Yep, was just about to make that change!

Reply:
`HodlUnsubscribeAll` will also only remove the queue from teh series of map indexes also. 

Reading your comment again, I think you mean unsubscribe, then stop the queue?

Reply:
That would match the behavior of the link: https://github.com/lightningnetwork/lnd/blob/a1b0a7eb027ce84a05360475164e95daf201796d/htlcswitch/link.go#L490-L506

Reply:
> Reading your comment again, I think you mean unsubscribe, then stop the queue?

Yeah, sorry, wanted to say "stop the queue _after_ unsubscribing".